src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -803,13 +803,35 @@ private bool IsValidExtensionMethodSignature()
             }
         }
 
-        private bool IsValidUserDefinedOperatorSignature(int parameterCount) =>
-                !this.ReturnsVoid &&
-                !this.IsGenericMethod &&
-                !this.IsVararg &&
-                this.ParameterCount == parameterCount &&
-                this.ParameterRefKinds.IsDefault && // No 'ref' or 'out'
-                !this.IsParams();
+        private bool IsValidUserDefinedOperatorSignature(int parameterCount)
+        {
+            if (this.ReturnsVoid || this.IsGenericMethod || this.IsVararg || this.ParameterCount != parameterCount || this.IsParams())
```

Comment:
Please enter a bug on rejecting ref-returning operators (defined in IL). It seems we have always allowed them, so there could be compat concerns.
Also it is not clear whether they "just worked", or were broken anyways.

Reply:
Filed #23606 for that.

---
In reply to: [154480103](https://github.com/dotnet/roslyn/pull/23508#discussion_r154480103) [](ancestors = 154480103)